person.rb

```diff
@@ -8,15 +10,20 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = Corrector.new
   end
 
-  def of_age?
-    @age >= 18
+  def can_use_services?
+    of_age? || @parent_permission == true
   end
 
-  private :of_age?
+  def validate_name
+    @corrector.correct_name @name
```

Comment:
- [ ] Please kindly use the Corrector instance to validate and save the corrected name e.g. `@name = @corrector.correct_name(@name)`

Reply:
Yes, sure.